### PR TITLE
[Merged by Bors] - feat(Set/Lattice): add `Set.seq_eq_image2`, use it to golf some proofs

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Lattice.lean [hunk 1]
```diff
@@ -2024,14 +2025,10 @@ theorem image2_iUnion_right (s : Set α) (t : ι → Set β) :
   simp only [← image_prod, prod_iUnion, image_iUnion]
 #align set.image2_Union_right Set.image2_iUnion_right
 
-/- ./././Mathport/Syntax/Translate/Expr.lean:107:6: warning: expanding binder group (i j) -/
-/- ./././Mathport/Syntax/Translate/Expr.lean:107:6: warning: expanding binder group (i j) -/
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Lattice.lean:2028]: These might be worth keeping as a reminder that `⋃ (i) (j), f i j` syntax is ugly and we still can't write the Lean 3 `⋃ i j, f i j`; or alternatively, we should just switch to `⋃ i, ⋃ j, f i j` below, if we think that's nicer.

File: Mathlib/Data/Set/Lattice.lean [hunk 2]
```diff
@@ -2049,33 +2046,22 @@ theorem image2_iInter_subset_right (s : Set α) (t : ι → Set β) :
...
-theorem image2_eq_iUnion (s : Set α) (t : Set β) : image2 f s t = ⋃ (i ∈ s) (j ∈ t), {f i j} := by
-  simp_rw [← image_eq_iUnion, iUnion_image_left]
-#align set.image2_eq_Union Set.image2_eq_iUnion
-
-/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
 theorem prod_eq_biUnion_left : s ×ˢ t = ⋃ a ∈ s, (fun b => (a, b)) '' t := by
   rw [iUnion_image_left, image2_mk_eq_prod]
 #align set.prod_eq_bUnion_left Set.prod_eq_biUnion_left
 
-/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Lattice.lean:2078]: These are indeed fine to remove.

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Data/Set/Lattice.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Set/Lattice.lean]: Reverted.

**Pull request closed without merging**